### ISSUE TITLE
qt5: cross-compilation against qt5-5.10.1 fails

### ISSUE THREAD

**Issue by Hoshpak** [voidlinux/void-packages#11723]
I just tried to cross-compile kde5 for aarch64 on a local branch rebased on top of master up to commit 813ac8c88855ac47dc8ce6645a3165a4ebc0d4cc . qt5 built fine however the first package that get's build which was bluez-qt5 fails:

```
CMake Error at /usr/aarch64-linux-gnu/usr/lib64/cmake/Qt5Core/Qt5CoreConfig.cmake:15 (message):
  The imported target "Qt5::Core" references the file

     "/builddir/qt-everywhere-src-5.10.1/host/bin/qmake"

  but this file does not exist.  Possible reasons include:

  * The file was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and contained

     "/usr/aarch64-linux-gnu/usr/lib64/cmake/Qt5Core/Qt5CoreConfigExtras.cmake"

  but not all the files it references.

Call Stack (most recent call first):
  /usr/aarch64-linux-gnu/usr/lib64/cmake/Qt5Core/Qt5CoreConfigExtras.cmake:6 (_qt5_Core_check_file_exists)
  /usr/aarch64-linux-gnu/usr/lib64/cmake/Qt5Core/Qt5CoreConfig.cmake:172 (include)
  /usr/share/ECM/modules/ECMQueryQmake.cmake:1 (find_package)
  /usr/share/ECM/modules/ECMAddQch.cmake:268 (include)
  CMakeLists.txt:17 (include)


-- Configuring incomplete, errors occurred!
See also "/builddir/bluez-qt-5.43.0/build/CMakeFiles/CMakeOutput.log".
```

The file `usr/aarch64-linux-gnu/usr/lib/cmake/Qt5Core/Qt5CoreConfigExtras.cmake` which originates from qt5-devel now lists `/builddir/qt-everywhere-src-5.10.1/host/bin/` as the path for the binaries. I compared that with qt5-5.8.0 and there the binaries are listed in `/usr/lib/qt5/bin/` where they actually are. 

CC @pullmoll

**Comment by pullmoll** [comment]
Damn. The replacing taking place in `_cleanup_wrksrc_leak()` obviously needs to be modified.
This means qt5-5.10.1_4 ...

**Comment by pullmoll** [comment]
I'll be testing this locally before commiting yet another update.
The builders are either full or awfully slow and don't have (enough) ccache it seems.

**Comment by pullmoll** [comment]
If you want to try the fix as well, here's the diff:
```
diff --git a/srcpkgs/qt5/template b/srcpkgs/qt5/template
index da0b127349..834d9cbf30 100644
--- a/srcpkgs/qt5/template
+++ b/srcpkgs/qt5/template
@@ -51,13 +51,13 @@ _cleanup_wrksrc_leak() {
        if [ -d "${PKGDESTDIR}/usr/lib/cmake" ]; then
                # Replace references to ${wrksrc} in cmake files
                sed -i ${PKGDESTDIR}/usr/lib/cmake/*/*.cmake \
-                       -e "s;${wrksrc}/qtbase/host;/usr/lib/qt5;g" \
+                       -e "s;${wrksrc}/host;/usr/lib/qt5;g" \
                        -e "s;devices/void-${XBPS_CROSS_TRIPLET}-g++;linux-g++;g"
        fi
        if [ -d "${PKGDESTDIR}/usr/lib/pkgconfig" ]; then
                # Replace references to ${wrksrc} in pkgconfig files
                sed -i ${PKGDESTDIR}/usr/lib/pkgconfig/*.pc \
-                       -e "s;${wrksrc}/qtbase/host;/usr/lib/qt5;g" \
+                       -e "s;${wrksrc}/host;/usr/lib/qt5;g" \
                        -e "s;devices/void-${XBPS_CROSS_TRIPLET}-g++;linux-g++;g"
        fi
        # Remove QMAKE_PRL_BUILD_DIR from hint files for static libraries
```

**Comment by Hoshpak** [comment]
Thanks, I will give it a try.

**Comment by Hoshpak** [comment]
I recompiled qt5 twice (for host and target), the resulting files look fine and I have successfully cross-compiled several KDE5 packages for aarch64 in the meantime.

**Comment by pullmoll** [comment]
Ok, my machine decided to stop building when the screen went black. Dunno why, something seems to have changed in the energy saving settings or whatever :p

**Comment by Hoshpak** [comment]
Fixed by 20dd327fab668b3869e53a73d1c4b553e3ccf1fd .
